scripts/test-api-endpoints.sh

```
@@ -0,0 +1,256 @@
+    local curl_cmd="curl -s -w '%{http_code}' -o /tmp/response.json"
+
+    if [ "$method" = "POST" ] || [ "$method" = "PUT" ] || [ "$method" = "PATCH" ]; then
+        curl_cmd="$curl_cmd -H 'Content-Type: application/json'"
+        if [ -n "$data" ]; then
+            curl_cmd="$curl_cmd -d '$data'"
+        fi
+    fi
+
+    curl_cmd="$curl_cmd -X $method '$BASE_URL$endpoint'"
+
+    local response_code=$(eval $curl_cmd)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)